src/dune/build_system.ml

```diff
@@ -470,6 +512,13 @@ let compute_targets_digest targets =
   | l -> Some (Digest.generic l)
   | exception (Unix.Unix_error _ | Sys_error _) -> None
 
+(* Find the closest corresponding source directory, eg
+   '_build/default/src/dune/.dune.objs' becomes 'src/dune' *)
+let find_rule_source_dir rule =
```

Comment:
It seems to me that this function should live in `rule.ml` under the name `Rule.find_source_dir`. (Also, the comment is a bit confusing since the input is not a path but a rule.)